src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9337,69 +9352,154 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                         memberGroup: candidates,
                         typeContainingConstructor: null,
                         delegateTypeBeingInvoked: null);
-                }
 
-                ImmutableArray<BoundExpression> arguments = BuildArgumentsForErrorRecovery(analyzedArguments, candidates);
+                    ImmutableArray<BoundExpression> arguments = BuildArgumentsForErrorRecovery(analyzedArguments, candidates);
 
-                // A bad BoundIndexerAccess containing an ErrorPropertySymbol will produce better flow analysis results than
-                // a BoundBadExpression containing the candidate indexers.
-                PropertySymbol property = (candidates.Length == 1) ? candidates[0] : CreateErrorPropertySymbol(candidates);
+                    // A bad BoundIndexerAccess containing an ErrorPropertySymbol will produce better flow analysis results than
+                    // a BoundBadExpression containing the candidate indexers.
+                    PropertySymbol property = (candidates.Length == 1) ? candidates[0] : CreateErrorPropertySymbol(candidates);
 
-                propertyAccess = BoundIndexerAccess.ErrorAccess(
-                    syntax,
-                    receiver,
-                    property,
-                    arguments,
-                    argumentNames,
-                    argumentRefKinds,
-                    candidates);
+                    propertyAccess = BoundIndexerAccess.ErrorAccess(
+                        syntax,
+                        receiver,
+                        property,
+                        arguments,
+                        analyzedArguments.GetNames(),
+                        analyzedArguments.RefKinds.ToImmutableOrNull(),
+                        candidates);
+                }
             }
             else
             {
-                MemberResolutionResult<PropertySymbol> resolutionResult = overloadResolutionResult.ValidResult;
-                PropertySymbol property = resolutionResult.Member;
+                propertyAccess = BindSuccessfulIndexerOrIndexedPropertyAccess(receiver, analyzedArguments, overloadResolutionResult, syntax, diagnostics);
+            }
+
+            overloadResolutionResult.Free();
+            return propertyAccess;
+        }
+
+        private bool TryBindExtensionIndexerAccess(SyntaxNode syntax, BoundExpression receiver,
+            AnalyzedArguments analyzedArguments, BindingDiagnosticBag diagnostics,
+            [NotNullWhen(true)] out BoundIndexerAccess? extensionIndexerAccess)
+        {
+            Debug.Assert(receiver.Type is not null);
+
+            if (receiver.Kind == BoundKind.BaseReference
+                || receiver.Type.IsTypeParameter() // PROTOTYPE need to confirm what we want for extension indexer access on type parameters
```

Comment:
>receiver.Type.IsTypeParameter() // PROTOTYPE need to confirm what we want for extension indexer access on type parameters [](http://example.com/codeflow?start=19&length=121)

It doesn't look like the condition is checking for a type parameter, I.e. the receiver isn't necessary a `T`, it could be a value of type `T`. However, it is not clear from the comment whether these scenarios should be treated differently.